### PR TITLE
Adds backblast element, gives it to the rocket launcher

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
The PM9 rocket launcher is a really sad little thing, with its standard rockets doing a paltry 30 damage on impact and a (0, 0, 2, 4) explosion (no fire even) when hitting a living mob. This means even a direct hit may not be enough to crit an unarmored target. The HEDP is a little bit better, but it's still nowhere fun enough for a ROCKET LAUNCHER. So I finally got around to an old promise and wrote up a backblast element, and gave it to the rocket launcher. I also made the standard rockets for the rocket launcher do a bit more impact damage and have some flames in their explosion, to make them less terribly sad.

[](https://i.imgur.com/l53U6aX.png)
_A screenshot of a backblast cloud midway through propagating, just after firing to the upper right_

This element can be attached to guns and causes a/several plumes of fire to go shooting out behind you when you fire said gun. These plumes of fire will bounce off any solid surface and burn any poor sap caught in its bullrush, as well as knocking any loose items left nearby away at high speeds. You can take a look at a WIP example here:

https://cdn.discordapp.com/attachments/326831214667235328/753059098550993056/backblast_Trim.mp4

I still have some neatening up and embellishing to do with the specifics of the plumes and the item knockbacks, and I definitely need to do some tweaking to the balance to make sure the backblast finds a nice level of power that makes it fun to use without being horribly OP. Stay tuned!

## Why It's Good For The Game
Gives nuke ops a reason to buy the rocket launcher, as well as a potential high-risk high-reward strategy of running up to a group of soft targets and firing directly away from them to scald them all with the backblast. Makes the rocket launcher more fun in general.

## Changelog
:cl: Ryll/Shaps
add: The rocket launcher now creates a giant backblast of fire and pressure when fired, be sure that you don't shoot it off from a corner (unless you're fireproof)!
/:cl:

### CODE REVIEW

File: code/modules/atmospherics/environmental/LINDA_fire.dm [hunk 1]
```diff
@@ -240,6 +240,15 @@
 		var/mob/living/L = AM
 		L.fire_act(temperature, volume)
 
+/obj/effect/hotspot/backblast
+	volume = 250
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/environmental/LINDA_fire.dm:244]: Why double the volume?

File: code/modules/atmospherics/environmental/LINDA_fire.dm [hunk 2]
```diff
@@ -240,6 +240,15 @@
...
+
+/obj/effect/hotspot/backblast/Crossed(atom/movable/AM, oldLoc)
+	..()
+	if(isliving(AM))
+		var/mob/living/L = AM
+		L.adjust_fire_stacks(5) // +5 firestacks from normal
+
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/environmental/LINDA_fire.dm:251]: and this is just to make it more spooky I assume?

**Comment by LemonInTheDark** [comment]
Unimaginably soulful, I do want you to know my pain tho, each of those green turfs is active, and being processed by the atmos ss.

If linda worked it wouldn't be a big deal, since heat alone settles quite fast. Know my pain.

**Comment by itseasytosee** [comment]
Seems a bit excessive huh.

**Comment by Ryll-Ryll** [comment]
The size of the fireball in the video is extra large to demonstrate how it works, as I said the numbers will be tuned

**Comment by wesoda25** [comment]
Jesus save some good ideas for the rest of us.

File: code/modules/atmospherics/environmental/LINDA_fire.dm (hunk 1 above)
Comment (Ryll-Ryll) [reply code/modules/atmospherics/environmental/LINDA_fire.dm]: undone

File: code/modules/atmospherics/environmental/LINDA_fire.dm (hunk 2 above)
Comment (Ryll-Ryll) [reply code/modules/atmospherics/environmental/LINDA_fire.dm]: yeah, and i lowered the number to +3

**Comment by Ryll-Ryll** [comment]
fireball smaller now

**Comment by Timberpoes** [comment]
I'm Commander Timberpoes and this is my favourite PR on the citadel.

Recoiless Rifle when?

File: code/datums/elements/backblast.dm [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+/datum/element/backblast/Detach(datum/source, force)
+	if(source)
+		UnregisterSignal(source, COMSIG_GUN_FIRED)
+	. = ..()
+
+/// For firing multiple plumes behind us, we evenly spread out our projectiles based on the [angle_spread][/datum/element/backblast/var/angle_spread] and [number of plumes][/datum/element/backblast/var/plumes]
+/datum/element/backblast/proc/gun_fired(obj/item/gun/weapon, mob/living/user, atom/target, params, zone_override)
+	SIGNAL_HANDLER
+
+	if(!weapon.chambered || HAS_TRAIT(user, TRAIT_PACIFISM))
```
Comment (ATH1909) [inline code/datums/elements/backblast.dm:41]: what about golems and such, or not meeting the firing pin requirements of the gun?

File: code/modules/projectiles/projectile/special/rocket.dm [hunk 4]
```diff
@@ -31,22 +32,24 @@
...
 	embedding = null
 
 /obj/projectile/bullet/a84mm_he/on_hit(atom/target, blocked=0)
 	..()
 	if(!isliving(target)) //if the target isn't alive, so is a wall or something
-		explosion(target, 0, 1, 2, 4)
+		explosion(target, 0, 1, 2, 4, flame_range = 3)
 	else
-		explosion(target, 0, 0, 2, 4)
+		explosion(target, 0, 1, 2, 4, flame_range = 3)
```
Comment (ATH1909) [inline code/modules/projectiles/projectile/special/rocket.dm:49]: is the flame range added onto the light range, or will the flames of this explosion not extend beyond its light explosion range?

**Comment by Tharcoonvagh** [comment]
SURRENDER NOW MAGGOTS AND YOU WILL NOT BE HARMED

**Comment by Ryll-Ryll** [comment]
Finish later

**Comment by LemonInTheDark** [comment]
:(

**Comment by Couls** [comment]
:(

**Pull request closed without merging**